inspire/modules/deposit/views.py

```diff
@@ -25,3 +42,79 @@
     template_folder='templates',
     static_folder="static",
 )
+
+
+CHART_TYPES = {'column': 'Column',
+               'pie': 'Pie'}
+
+
+class FilterDateForm(Form):
+
+    """Date form."""
+
+    since_date = DateTimeField(
+        label=_('From'),
+        description='Format: YYYY-MM-DD.')
+    until_date = DateTimeField(
+        label=_('To'),
+        description='Format: YYYY-MM-DD.')
+    submit = SubmitField(_("Filter depositions"))
+
+
+@deposit_blueprint.route('/<depositions:deposition_type>/stats',
+                         methods=['GET'])
+@login_required
+def show_stats(deposition_type):
+    """Render the stats for all the depositions."""
+    if len(DepositionType.keys()) <= 1 and \
+       DepositionType.get_default() is not None:
+        abort(404)
+
+    form = FilterDateForm()
+
+    deptype = DepositionType.get(deposition_type)
+    submitted_depositions = [d for d in Deposition.get_depositions(type=deptype) if d.has_sip(sealed=True)]
+
+    ctx = process_metadata_for_charts(submitted_depositions,
+                                      group_by=request.args.get('group_by', 'type_of_doc'))
+    ctx.update(dict(
+        deposition_type=deptype,
+        depositions=submitted_depositions,
+        form=form,
+        chart_types=CHART_TYPES
+    ))
+
+    # Send signal to allow modifications to the template context
+    template_context_created.send(
```

Comment:
I don't think this is needed, please check.

We don't have any listener for this signal so nothing will happen. This was probably done for Zenodo case.

See here how the signal is connected:

https://github.com/zenodo/zenodo/blob/4d4823a38adc5c46b07e21b4b70ec767399f3054/zenodo/modules/deposit/views.py#L45

And here is the handler:

https://github.com/zenodo/zenodo/blob/4d4823a38adc5c46b07e21b4b70ec767399f3054/zenodo/modules/deposit/receivers.py#L43